Code/Framework/AzToolsFramework/AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.cpp

```diff
@@ -2651,6 +2652,17 @@ namespace AzToolsFramework
                 {
                     AZ_PROFILE_FUNCTION(AzToolsFramework);
 
+                    // Don't allow users to delete entities while in Entity Picker mode.
+                    if (auto viewportEditorModeTracker = AZ::Interface<AzToolsFramework::ViewportEditorModeTrackerInterface>::Get())
+                    {
+                        auto viewportEditorModes =
+                            viewportEditorModeTracker->GetViewportEditorModes({ AzToolsFramework::GetEntityContextId() });
+                        if (viewportEditorModes->IsModeActive(AzToolsFramework::ViewportEditorMode::Pick))
+                        {
+                            return;
+                        }
+                    }
```

Comment:
suggestion: It might be worth pulling this out into a simple local function/lambda with`auto` and then capturing it in the callbacks, e.g.

```
auto isInEditorPickMode = []
{
    // Don't allow users to delete entities while in Entity Picker mode.
    if (auto viewportEditorModeTracker = AZ::Interface<AzToolsFramework::ViewportEditorModeTrackerInterface>::Get())
    {
        auto viewportEditorModes =
            viewportEditorModeTracker->GetViewportEditorModes({ AzToolsFramework::GetEntityContextId() });
        if (viewportEditorModes->IsModeActive(AzToolsFramework::ViewportEditorMode::Pick))
        {
            return true;
        }
    }
    return false;
}
...

[this, isInEditorPickMode]()
{
    AZ_PROFILE_FUNCTION(AzToolsFramework);
    if (isInEditorPickMode())
    {
        return;
    }
    ...
```